hamilton/plugins/jupyter_magic.py

```diff
@@ -163,95 +154,154 @@ def determine_notebook_type() -> str:
 
 @magics_class
 class HamiltonMagics(Magics):
-    """Magics to facilitate Hamilton development in Jupyter notebooks"""
+    """Magics to facilitate interactive Hamilton development in notebooks."""
+
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+        self.builder = None
+        self.notebook_env = determine_notebook_type()
 
     @magic_arguments()  # needed on top to enable parsing
+    @argument("module_name", help="Name for the module defined in this cell.")
+    @argument(
+        "-d",
+        "--display",
+        nargs="?",
+        const=True,
+        help="Display the dataflow. The argument is the variable name of a dictionary of visualization kwargs; else {}.",
+    )
     @argument(
-        "-m", "--module_name", help="Module name to provide. Default is jupyter_module."
-    )  # keyword / optional arg
+        "-x",
+        "--execute",
+        nargs="?",
+        const=True,
+        help="Execute the dataflow. The argument is the variable name of a list of nodes; else execute all nodes.",
+    )
     @argument(
-        "-c", "--config", help="JSON config, or variable name containing config to use."
```

Comment:
we should keep this for backwards compatibility. 
If someone provides a builder we should error, or replace the builder's config with this.